src/map_command_handler.hpp

```diff
@@ -190,9 +190,30 @@ class map_command_handler
 		}
 		else if (help_on_unknown_) {
 			utils::string_map symbols;
+		    // Get the input command on a string
+            std::string string_user_ = get_cmd();
+            // Get all commands (even if disallowed)
+            std::vector<std::string> command_list_ = get_commands_list();
+            // Initialize the distance and the command_proposal symbol so it
+            // does not appear if no candidate is found.
+            symbols["command_proposal"] = "";
+            int distance_ = 0;
+            // Compare the input with every command (excluding alias)
+            for (typename command_map::value_type i : command_map_) {
+                // No need to test commands that are not enabled
+                if(is_enabled(i.second)) {
+                    distance_ = approximate_string_distance(string_user_, i.first);
+                    // Maximum of a third of the letters are wrong
+                    if (distance_*100/std::min(string_user_.length(),i.first.length()) < 34){
```

Comment:
Again, please add spaces around the operators.

It may also make it more readable to separate the calculation out into steps.

Reply:
> It may also make it more readable to separate the calculation out into steps.

I dont really know what you mean by separating the calculation into steps... Do you mean, for example, putting `std::min(string_user_.length(),i.first.length())` into a variable first, and then using that variable in the calculation?

Reply:
Yeah, that's basically what I mean.

Reply:
Can do.